### PR TITLE
KOTOR2: Character generation menu

### PR DESCRIPTION
**Pull request by Nostritius**
This PR implements the kotor 2 character generation menu. It is in many cases very similar to the kotor character generation menu.

### CODE REVIEW

File: src/engines/kotor2/gui/chargen/chargenname.h [hunk 1]
```diff
@@ -28,17 +28,25 @@
 #include "src/engines/aurora/console.h"
 #include "src/engines/kotor2/gui/chargen/chargenbase.h"
 #include "src/engines/kotor2/gui/chargen/chargeninfo.h"
+#include "src/engines/kotor2/gui/widgets/label.h"
 
 namespace Engines {
 
 namespace KotOR2 {
 
+class WidgetLabel;
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargenname.h]: Yeah, that doesn't work. You can't forward-declare that fake "proxy" class.

You need to either forward-declare Engines::KotOR::WidgetLabel, include "src/engines/kotor2/gui/widgets/label.h" (which you have here, but then removed again) and not forward-declare or do it inline with the definition below, i.e. "class ::Engines::KotOR::WidgetLabel *_nameLabel;"

File: src/engines/kotor2/gui/chargen/chargeninfo.cpp [hunk 2]
```diff
@@ -237,6 +341,57 @@ Common::UString CharacterGenerationInfo::getPortrait() const {
...
+			}
+		case kSkinB:
+			if (_gender == kGenderFemale) return _face + 1;
+			else return _face + 6;
+		case kSkinC:
+			switch (_face) {
+				case 0: return 1;
+				case 1:
+					if (_gender == kGenderFemale) return 2;
+					else if (_gender == kGenderMale) return 3;
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargeninfo.cpp]: You forgot the throw here

File: src/engines/kotor2/gui/chargen/rules.mk [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# You should have received a copy of the GNU General Public License
+# along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+
+# Character Generation system in Star Wars: Knights of the Old Republic II - The Sith Lords.
+
+src_engines_kotor2_libkotor2_la_SOURCES += \
+    $(EMPTY)
+
+src_engines_kotor2_libkotor2_la_SOURCES += \
+    $(EMPTY)
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/rules.mk]: Missing newline at the end of the file

File: src/engines/kotor2/gui/chargen/classselection.cpp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	_guardianFemaleButton = getButton("BTN_SEL5");
+	_consularFemaleButton = getButton("BTN_SEL6");
+
+	_labelDesc->setText(_consularDescription);
+	_labelTitle->setText(_consularMaleTitle);
+	_hoveredButton = _consularMaleButton;
+}
+
+void ClassSelection::callbackRun()
+{
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/classselection.cpp]: That `{` belongs onto the previous line

File: src/engines/kotor2/gui/chargen/classselection.cpp [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	} else if (_guardianFemaleButton->isHovered() && _hoveredButton != _guardianFemaleButton) {
+		_labelDesc->setText(_guardianDescription);
+		_labelTitle->setText(_guardianFemaleTitle);
+		_hoveredButton = _guardianFemaleButton;
+	}
+}
+
+void ClassSelection::callbackActive(Widget &widget)
+{
+	//TODO add different character generations
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/classselection.cpp]: `// TODO: Add`

File: src/engines/kotor2/gui/chargen/chargeninfo.cpp [hunk 6]
```diff
@@ -0,0 +1,123 @@
+//
+// Created by patrick on 22.10.17.
+//
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargeninfo.cpp]: Use our standard header, please

File: src/engines/kotor2/gui/chargen/chargeninfo.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
+//
+// Created by patrick on 22.10.17.
+//
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargeninfo.h]: Standard header, please.

File: src/engines/kotor2/gui/chargen/chargeninfo.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+	CharacterGenerationInfo();
+};
+
+} // End of namespace KotOR2
+
+} // End of namespace Engines
+
+
+#endif // ENGINES_KOTOR2_GUI_CHARGEN_CHARACTERGENERATIONINFO_H
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargeninfo.h]: Missing newline at end of file

File: src/engines/kotor2/gui/chargen/charactergeneration.cpp [hunk 9]
```diff
@@ -43,14 +43,16 @@ CharacterGeneration::CharacterGeneration(Module *module, CharacterGenerationInfo
 			getLabel("LBL_CLASS")->setText(TalkMan.getString(355));
 			break;
 		case kClassJediSentinel:
-			getLabel("LBL_CLASS")->setText(TalkMan.getString(356));
+			getLabel("LBL_CLASS")->setText(TalkMan.getString(353));
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/charactergeneration.cpp]: Can you put that fix into the previous commit, as if that mistake never happened?

File: src/engines/kotor2/gui/chargen/chargenbase.cpp [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+#include "src/engines/kotor2/gui/chargen/chargenbase.h"
+
+namespace Engines {
+
+namespace KotOR2 {
+
+CharacterGenerationBaseMenu::CharacterGenerationBaseMenu(CharacterGenerationInfo &info,
+														 Engines::Console *console) : GUI(console), _info(info), _accepted(false) {
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargenbase.cpp]: Please use spaces for alignment here.

File: src/engines/kotor2/gui/chargen/chargenname.cpp [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "src/engines/kotor2/gui/widgets/kotorwidget.h"
+#include "src/engines/kotor2/gui/chargen/chargenname.h"
+
+namespace Engines {
+
+namespace KotOR2 {
+
+CharacterGenerationNameMenu::CharacterGenerationNameMenu(CharacterGenerationInfo &info, Engines::Console *console) :
+	CharacterGenerationBaseMenu(info, console) {
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargenname.cpp]: That needs to be indented one level further.

File: src/engines/kotor2/gui/chargen/chargenportrait.cpp [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+#include "src/engines/kotor2/gui/widgets/kotorwidget.h"
+
+#include "src/engines/kotor2/gui/chargen/chargenportrait.h"
+
+namespace Engines {
+
+namespace KotOR2 {
+
+CharacterGenerationPortraitMenu::CharacterGenerationPortraitMenu(CharacterGenerationInfo &info,
+																 Engines::Console *console) :
```
Comment (DrMcCoy) [inline src/engines/kotor2/gui/chargen/chargenportrait.cpp]: Spaces for alignment

**Comment by Nostritius** [comment]
The issues should now be fixed

**Comment by DrMcCoy** [comment]
Merged, thanks! :)

**Pull request merged**